core: enabling logCollector by default for coredump collection (#11163)
rook/rook: Documentation/CRDs/Cluster/ceph-cluster-crd.md

  * `disable`: is set to `true`, the crash collector will not run on any node where a Ceph daemon runs
  * `daysToRetain`: specifies the number of days to keep crash entries in the Ceph cluster. By default the entries are kept indefinitely.
* `logCollector`: The settings for log collector daemon.

>   * `enabled`: if set to `true`, the log collector will run as a side-car next to each Ceph daemon. The Ceph configuration option `log_to_file` will be turned on, meaning Ceph daemons will log on files in addition to still logging to container's stdout. These logs will be rotated. (default: true)

Reviewer: ```suggestion
  * `enabled`: if set to `true`, the log collector will run as a side-car next to each Ceph daemon. The Ceph configuration option `log_to_file` will be turned on, meaning Ceph daemons will log on files in addition to still logging to container's stdout. These logs will be rotated. (default: `true`)
```

>   * `periodicity`: how often to rotate daemon's log. (default: 24h). Specified with a time suffix which may be 'h' for hours or 'd' for days. **Rotating too often will slightly impact the daemon's performance since the signal briefly interrupts the program.**

Reviewer: ```suggestion
  * `periodicity`: how often to rotate daemon's log. (default: `24h`). Specified with a time suffix which may be `h` for hours or `d` for days. **Rotating too often will slightly impact the daemon's performance since the signal briefly interrupts the program.**
```

  * `disable`: is set to `true`, the crash collector will not run on any node where a Ceph daemon runs
  * `daysToRetain`: specifies the number of days to keep crash entries in the Ceph cluster. By default the entries are kept indefinitely.
* `logCollector`: The settings for log collector daemon.

>   * `enabled`: if set to `true`, the log collector will run as a side-car next to each Ceph daemon. The Ceph configuration option `log_to_file` will be turned on, meaning Ceph daemons will log on files in addition to still logging to container's stdout. These logs will be rotated. The coredump files will be generated in `/var/lib/systemd/coredump` directory on the host where the pod is running in case a daemon terminates with a segfault. (default: `true`)

Reviewer: Are we sure this is happening for all underlying OSes/kernels? What happens when the `kernel.core_pattern` is not pointing to the `coredumpctl`/ just a path/directory.

Author: @galexrt @travisn please let me know if the new wording sounds good based on our discussion or we can make it better ?

>   * `enabled`: if set to `true`, the log collector will run as a side-car next to each Ceph daemon. The Ceph configuration option `log_to_file` will be turned on, meaning Ceph daemons will log on files in addition to still logging to container's stdout. These logs will be rotated. The coredump files will be generated in `/var/lib/systemd/coredump` directory on the host depending on the underlying OS location for coredumps where the pod is running in case a daemon terminates with a segfault. (default: `true`)

Reviewer 2: Looks good, just a small suggestion
```suggestion
  * `enabled`: if set to `true`, the log collector will run as a side-car next to each Ceph daemon. The Ceph configuration option `log_to_file` will be turned on, meaning Ceph daemons will log on files in addition to still logging to container's stdout. These logs will be rotated. In case a daemon terminates with a segfault, the coredump files will be commonly be generated in `/var/lib/systemd/coredump` directory on the host, depending on the underlying OS location. (default: `true`)
```

Author: sure, updated.
